### ISSUE TITLE
[question] should we model `gender` like a vocabulary

### ISSUE THREAD

**Issue by stefanprobst** [acdh-oeaw/InTaVia-Backend#17]
- we may want to have the `Person.gender` *field* in similar shape as `occupations`, i.e. `{ id: string; label: InternationalizedLabel }`
- as a *query param*, we may want to allow `gender?: Array<Gender[id]>`

**Comment by sennierer** [comment]
yes, good point

**Comment by stefanprobst** [comment]
also needs to be changed for the `gender` query param to accept any gender id, not the 3-value enum?

**Comment by sennierer** [comment]
thats true. due to consistency with the other params I'll add a separate gender_id param

**Comment by sennierer** [comment]
fixed in 206d5db97f37ad443c4dd0ea3490d8c41282569b; wrong issue id in the commit message

**Comment by stefanprobst** [comment]
still confused by this:
![Screenshot_20220722_084338](https://user-images.githubusercontent.com/20753323/180379777-ac26e5f3-48df-4a7a-8308-a5861cc54718.png)

where does the enum apply?

**Comment by sennierer** [comment]
the enum is still accepted by the gender parameter, the gender_id param accepts any gender_id (any URI as used in the triplestore), eg http://ldf.fi/schema/bioc/Male

**Comment by stefanprobst** [comment]
so what happens when a query provides `?gender=female` - what field gets searched? `gender.label`?

**Comment by sennierer** [comment]
basically yes, but technically the enum is a map between uris and strings which is hardcoded in the backend server. So, the sparql uses the URI that is mapped to female, but `URI -> rdfs.label == "female"`. This is probably not ideal as in case of changes the labels need to be changed in the data and the server. I'll give that another thought

**Comment by sennierer** [comment]
No. That enum gets translated into the correct Uri and that is used in the sparql query. But speaking from the Json return Format, yes its gender.label

Am 22. Juli 2022 10:02:45 MESZ schrieb stefanprobst ***@***.***>:
